examples/module1/lesson1/solver/App.tsx

```diff
@@ -1,58 +1,54 @@
+import Button from './components/Button/Button';
+import Input from './components/Input/Input';
+import { calculate } from './calculate';
+import { Operations } from './operations';
 
 const App = () => {
-  const [numA, setNumA] = useState<number>(0);
-  const [numB, setNumB] = useState<number>(0);
-  const [numC, setNumC] = useState<number | string>(0);
+  const [firstNumber, setFirstNumber] = useState<number>(0);
+  const [secondNumber, setSecondNumber] = useState<number>(0);
+  const [result, setResult] = useState<number | null>(null);
+  const [error, setError] = useState<string | null>(null);
 
-  const doWork = (func: (a: number, b: number) => number) => {
-    setNumC(func(numA, numB));
+  const handleCalculation = (operation: Operations) => () => {
+    try {
+      const calcResult: number | string = calculate({
+        firstNumber,
+        secondNumber,
+        operation,
+      });
+      setResult(calcResult);
+      setError(null); // Clear previous errors
+    } catch (err) {
+      setError((err as Error).message);
+      setResult(null); // Clear previous result
+    }
   };
 
   return (
     <div>
       <div className="grid grid-cols-2 gap-x-4">
-        <input
+        <Input
           type="number"
-          className="rounded-md shadow-md p-4"
-          value={numA}
-          onChange={(e) => setNumA(parseFloat(e.target.value))}
+          value={firstNumber}
+          onChange={(e) => setFirstNumber(parseFloat(e.target.value))}
         />
-        <input
+        <Input
           type="number"
-          className="rounded-md shadow-md p-4"
-          value={numB}
-          onChange={(e) => setNumB(parseFloat(e.target.value))}
+          value={secondNumber}
+          onChange={(e) => setSecondNumber(parseFloat(e.target.value))}
         />
       </div>
       <div className="grid grid-cols-4 gap-x-4 my-4">
-        <button
-          className="bg-blue-200 px-2 py-4 text-lg hover:bg-blue-500 hover:text-white rounded-md"
-          onClick={() => doWork(f1)}
-        >
-          +
-        </button>
-        <button
-          className="bg-blue-200 px-2 py-4 text-lg hover:bg-blue-500 hover:text-white rounded-md"
-          onClick={() => doWork(f2)}
-        >
-          -
-        </button>
-        <button
-          className="bg-blue-200 px-2 py-4 text-lg hover:bg-blue-500 hover:text-white rounded-md"
-          onClick={() => doWork(f3)}
-        >
+        <Button onClick={handleCalculation(Operations.ADDITION)}>+</Button>
+        <Button onClick={handleCalculation(Operations.DIVISION)}>-</Button>
+        <Button onClick={handleCalculation(Operations.MULTIPLICATION)}>
           *
-        </button>
-        <button
-          className="bg-blue-200 px-2 py-4 text-lg hover:bg-blue-500 hover:text-white rounded-md"
-          onClick={() => doWork(f4)}
-        >
-          /
-        </button>
+        </Button>
+        <Button onClick={handleCalculation(Operations.SUBTRACTION)}>/</Button>
```

Comment:
tutaj to co w poprzednim komentarzu `SUBTRACTION` to odejmowanie a mamy `/` sugerującego dzielenie